CMake misdetecting presence of GPhoto2

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

Please:

* Read the documentation to test with the latest developer build.
* Check if other person has already created the same issue to avoid duplicates. You can comment on it if there already is an issue.
* Try to be as detailed as possible in your report.
* Report only one problem per created issue.


This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
- OpenCV => Current master (63564039640c81256ccfedf77899e37746352d45 on 2018-8-23)
- Operating System / Platform => Ubuntu 18.04
- Compiler => GCC 7.3.0, CMake version 3.10.2

##### Detailed description
When compiling OpenCV I ran into the following error:

    Scanning dependencies of target opencv_videoio
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/videoio_registry.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/videoio_c.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_images.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_mjpeg_encoder.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_mjpeg_decoder.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/container_avi.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_dc1394_v2.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_gstreamer.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_v4l.cpp.o
    [ 36%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_gphoto2.cpp.o
    /home/ruben/Projects/Libraries/opencv/modules/videoio/src/cap_gphoto2.cpp:32:10: fatal error: gphoto2/gphoto2.h: No such file or directory
     #include <gphoto2/gphoto2.h>
              ^~~~~~~~~~~~~~~~~~~

So it's missing the header files for gPhoto2. However, CMake says that it detects gPhoto2 to be present:

    --     gPhoto2:                     YES

Turns out that the problem was that I did have gPhoto2 present on my system, but not the development files that includes the header that OpenCV needs for this module. CMake detected gPhoto2, but this was a false positive as I didn't have the required files anyway. Installing Ubuntu's package for libgphoto2-dev fixed the problem for me, because that put `gphoto2.h` on the include path.

##### Steps to reproduce

1. Install all dependencies of OpenCV except gPhoto2.
2. Install the Ubuntu package for `libgphoto2-6`, but not `libgphoto2-dev`.
3. Build OpenCV using CMake.

##### Actual results

* Error at 36% that `gphoto2.h` is not found.

##### Expected results

gPhoto2 should not be marked as found in the configure stage of OpenCV, if its header files are not present. Furthermore, I'd expect one of these things to happen, whichever you think is most common behaviour for OpenCV:

* CMake disables whatever modules require gPhoto2 (videoio at least), or
* an error is raised during the configure stage that not all modules can be built as desired.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -276,7 +276,7 @@ OCV_OPTION(WITH_VA             "Include VA support"                          OFF
 OCV_OPTION(WITH_VA_INTEL       "Include Intel VA-API/OpenCL support"         OFF  IF (UNIX AND NOT ANDROID) )
 OCV_OPTION(WITH_MFX            "Include Intel Media SDK support"             OFF   IF ((UNIX AND NOT ANDROID) OR (WIN32 AND NOT WINRT AND NOT MINGW)) )
 OCV_OPTION(WITH_GDAL           "Include GDAL Support"                        OFF  IF (NOT ANDROID AND NOT IOS AND NOT WINRT) )
-OCV_OPTION(WITH_GPHOTO2        "Include gPhoto2 library support"             ON   IF (UNIX AND NOT ANDROID AND NOT IOS) )
+OCV_OPTION(WITH_GPHOTO2        "Include gPhoto2 library support"             OFF  IF (UNIX AND NOT ANDROID AND NOT IOS) )
 OCV_OPTION(WITH_LAPACK         "Include Lapack library support"              (NOT CV_DISABLE_OPTIMIZATION)  IF (NOT ANDROID AND NOT IOS) )
 OCV_OPTION(WITH_ITT            "Include Intel ITT support"                   ON   IF (NOT APPLE_FRAMEWORK) )
 OCV_OPTION(WITH_PROTOBUF       "Enable libprotobuf"                          ON )

